Scalafix, scalafmt and minor cleanups

jgogstad opened the pull request:
* Replace scalastyle with scalafmt
* Introduce scalafix
* Add scalafmt and scalafix checks to Travis build
* Use cats syntax instead of type class instance methods in `SftpStore`
  - Scalafix blew up on formatting this. It became a lot more readable with the syntax, especially the outer flatMaps.
* Minor stuff: Cleanup gitignore, add note to changelog that release notes goes to github release notes, move `sshd_config` from root directory to `sftp/src/test/resources`

gafiatulin commented on sftp/src/main/scala/blobstore/sftp/SftpStore.scala:
```diff
@@ -48,20 +49,18 @@ final class SftpStore[F[_]](
-    case ch if ch.isClosed => F.unit
-    case ch => F.ifM(mVar.tryPut(ch))(F.unit, SftpStore.closeChannel(semaphore, blocker)(ch))
+  private def channelResource: Resource[F, ChannelSftp] = Resource.make(getChannel) {
+    case ch if ch.isClosed => Applicative[F].unit
```
Consider `Applicative[F].unit` -> `().pure[F]` in vein of using syntax.
Two other occurrences in this file.

gafiatulin commented on sftp/src/main/scala/blobstore/sftp/SftpStore.scala:
```diff
@@ -19,24 +19,25 @@ package sftp
   semaphore: Option[Semaphore[F]],
   connectTimeout: Int
-)(implicit F: ConcurrentEffect[F], CS: ContextShift[F]) extends Store[F] {
+)(implicit CS: ContextShift[F]) extends Store[F] {
```
Consider keeping existing de-sugared syntax to be consistent with different strore types.
Same for the constructor in companion object.

jgogstad commented:
Addressed comments and some formatting mistakes, primarily from having multi-line blocks in parenthesis instead of curly brackets (see [b61f9af..854f3ea](https://github.com/fs2-blobstore/fs2-blobstore/compare/b61f9af05c38b2c3523eefd6efb3bf0d516f7549..854f3eae8dc7a37f6fc558e6ffe2935e3b24c461)).
